src/rgw/rgw_bucket.cc

```diff
@@ -1056,7 +1056,7 @@ int RGWBucket::check_bad_index_multipart(RGWBucketAdminOpState& op_state,
       int pos = oid.find_last_of('.');
       if (pos < 0) {
         /* obj has no suffix */
-        all_objs[key] = oid;
```

Comment:
you can remove the key variable and change rgw_obj obj(bucket, key) to  rgw_obj obj(bucket,  iter->key)